tests/xdc/history_replication_dlq_test.go

```diff
@@ -0,0 +1,454 @@
+		IsGlobalNamespace:                true,
+		WorkflowExecutionRetentionPeriod: timestamp.DurationPtr(1 * time.Hour * 24), // required param
+	})
+	s.NoError(err)
+
+	// Create a worker and register a workflow.
+	activeClient, err := sdkclient.Dial(sdkclient.Options{
+		HostPort:  s.cluster1.GetHost().FrontendGRPCAddress(),
+		Namespace: ns,
+	})
+	s.NoError(err)
+	tq := "dlq-replication-test-task-queue"
+	worker := sdkworker.New(activeClient, tq, sdkworker.Options{})
+	myWorkflow := func(ctx workflow.Context) (string, error) {
+		return "hello", nil
+	}
+	worker.RegisterWorkflow(myWorkflow)
+	s.NoError(worker.Start())
+	defer worker.Stop()
+
+	// Wait for the namespace to be replicated.
+	for {
+		var task *replicationspb.NamespaceTaskAttributes
+		select {
+		case task = <-s.namespaceReplicationTaskExecutorParams.tasks:
+		case <-ctx.Done():
+			s.FailNow("timed out waiting for namespace replication task to be processed")
+		}
+		if task.Info.Name == ns {
+			break
+		}
+	}
+
+	// Execute the workflow.
+	run, err := activeClient.ExecuteWorkflow(ctx, sdkclient.StartWorkflowOptions{
+		TaskQueue: tq,
+		ID:        workflowIDToFail,
+	}, myWorkflow)
+	s.NoError(err)
+
+	// Wait for the workflow to complete.
+	var result string
+	err = run.Get(ctx, &result)
+	s.NoError(err)
+	s.Equal("hello", result)
+
+	// Wait for the replication task executor to process the replication task.
+	select {
+	case <-s.replicationTaskExecutorParams.tasks:
+	case <-ctx.Done():
+		s.FailNow("timed out waiting for replication task to be processed")
+	}
+
+	// Wait for at least one replication task to be added to the DLQ. There could be more, but we only care about the
+	// first one because they should all be for this workflow.
+	var request *persistence.PutReplicationTaskToDLQRequest
+	select {
+	case request = <-s.executionManagerParams.dlqRequests:
+	case <-ctx.Done():
+		s.FailNow("timed out waiting for replication task to be added to DLQ")
+	}
+	s.Equal(s.clusterNames[0], request.SourceClusterName)
+	s.Equal(1, int(request.ShardID))
+
+	// Create a temporary file for TDBG output
+	file, err := os.CreateTemp("", "*")
+	s.NoError(err)
+	s.T().Cleanup(func() {
+		s.NoError(os.Remove(file.Name()))
+	})
+
+	// Create a TDBG client pointing at the standby cluster.
+	clientFactory := tdbg.NewClientFactory(
+		tdbg.WithFrontendAddress(s.cluster2.GetHost().FrontendGRPCAddress()),
+	)
+	app := tdbg.NewCliApp(clientFactory, tasks.NewDefaultTaskCategoryRegistry())
+	app.ExitErrHandler = func(c *cli.Context, err error) {
+		// We don't want the CLI to exit when it encounters an error, so we override the default handler.
+	}
+
+	// Run the TDBG command to read replication tasks from the DLQ.
+	// The last message ID is set to MaxInt64 - 1 because the last message ID is exclusive, so if it were
+	// set to the max int64, then, because the server increments the value, it would overflow and return no results.
+	lastMessageID := strconv.Itoa(math.MaxInt64 - 1)
```

Comment:
Could this be simplified to `:= fmt.Sprint(math.MaxInt32)`?

Reply:
I clarified in the comment why we want it to be as large as possible. It's possible for task ids to be > math.MaxInt32, so we could miss some tasks if we do that.

Reply:
Oh, so the task ids won't be sequential, starting from zero. That makes sense then 👍

Reply:
Yeah. Initially, I set this to 100K, but the tests were failing because the task IDs were actually larger than that even.